src/main/java/org/takes/rq/RqRequestLine.java

```diff
@@ -0,0 +1,159 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.rq;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.EqualsAndHashCode;
+import org.takes.HttpException;
+import org.takes.Request;
+
+/**
+ * HTTP Request-Line parsing.
+ *
+ * <p>All implementations of this interface must be immutable and thread-safe.
+ *
+ * @author Vladimir Maksimenko (xupypr@xupypr.com)
+ * @version $Id$
+ * @since 1.0
+ */
+public interface RqRequestLine extends Request {
+
+    /**
+     * METHOD token.
+     */
+    int METHOD = 1;
+
+    /**
+     * URI token.
+     */
+    int URI = 2;
+
+    /**
+     * HTTPVERSION token.
+     */
+    int HTTPVERSION = 3;
+
+    /**
+     * Get Request-Line header.
+     * @return HTTP Request-Line header
+     * @throws IOException If fails
+     */
+    String requestLineHeader() throws IOException;
+
+    /**
+     * Get Request-Line header token by index number.
+     * @param index Token index number
+     * @return HTTP Request-Line header token specified by index number
+     * @throws IOException If fails
+     */
+    String requestLineHeaderToken(int index) throws IOException;
+
+    /**
+     * Request decorator for Request-Line header validation
+     *
+     * <p>The class is immutable and thread-safe.
+     * @author Vladimir Maksimenko (xupypr@xupypr.com)
+     * @version $Id$
+     * @since 1.0
+     */
+    @EqualsAndHashCode(callSuper = true)
+    final class Base extends RqWrap implements RqRequestLine {
+        /**
+         * HTTP Request-line pattern.
+         * [!-~] is for method or extension-method token (octets 33 - 126).
+         * @see <a href="http://www
+         * .w3.org/Protocols/rfc2616/rfc2616-sec5.html#sec5.1">RFC 2616</a>
+         */
+        private static final Pattern PATTERN = Pattern.compile(
+                "([!-~]+) ([^ ]+)( [^ ]+)?"
+                );
+
+        /**
+         * Ctor.
+         * @param req Original request
+         */
+        public Base(final Request req) {
+            super(req);
+        }
+
+        @Override
+        public String requestLineHeader() throws IOException {
+            return this.extractFromRequestLineHeader(null);
```

Comment:
@xupyprmv can we avoid using null here and below?
http://www.yegor256.com/2014/05/13/why-null-is-bad.html